server/src/main/java/org/elasticsearch/index/query/QueryShardContext.java

```diff
@@ -119,18 +121,38 @@ public QueryShardContext(int shardId,
                              String clusterAlias,
                              Predicate<String> indexNameMatcher,
                              BooleanSupplier allowExpensiveQueries,
-                             ValuesSourceRegistry valuesSourceRegistry) {
-        this(shardId, indexSettings, bigArrays, bitsetFilterCache, indexFieldDataLookup, mapperService, similarityService,
-                scriptService, xContentRegistry, namedWriteableRegistry, client, searcher, nowInMillis, indexNameMatcher,
-                new Index(RemoteClusterAware.buildRemoteIndexName(clusterAlias, indexSettings.getIndex().getName()),
-                        indexSettings.getIndex().getUUID()), allowExpensiveQueries, valuesSourceRegistry);
+                             ValuesSourceRegistry valuesSourceRegistry,
+                             Map<String, Object> extraMappings) {
+        this(
+            shardId,
+            indexSettings,
+            bigArrays,
+            bitsetFilterCache,
+            indexFieldDataLookup,
+            mapperService,
+            similarityService,
+            scriptService,
+            xContentRegistry,
+            namedWriteableRegistry,
+            client,
+            searcher,
+            nowInMillis,
+            indexNameMatcher,
+            new Index(
+                RemoteClusterAware.buildRemoteIndexName(clusterAlias, indexSettings.getIndex().getName()),
+                indexSettings.getIndex().getUUID()
+            ),
+            allowExpensiveQueries,
+            valuesSourceRegistry,
+            mapperService.newFieldTypeLookup(extraMappings)
+        );
```

Comment:
although I see the value in reformatting like this as we go, I would try not to do it in a feature branch, because it will cause merge conflicts. Though I see how in this particular case, changing arguments here will anyway cause merge conflicts with master because things are not formatted nicely like you did.

Reply:
That was just a habit from the work I and been doing. I hadn't been thinking about the branch. I'll have another look on Monday.